aws_signing_helper/update.go

```diff
@@ -0,0 +1,200 @@
+package aws_signing_helper
+
+import (
+	"bufio"
+	"log"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+)
+
+const UpdateRefreshTime = time.Minute * time.Duration(5)
+const AwsSharedCredentialsFileEnvVarName = "AWS_SHARED_CREDENTIALS_FILE"
+
+// Structure to contain a temporary credential
+type TemporaryCredential struct {
+	AccessKeyId     string
+	SecretAccessKey string
+	SessionToken    string
+	Expiration      time.Time
+}
+
+// Updates credentials in the credentials file for the specified profile
+func Update(credentialsOptions CredentialsOpts, profile string, once bool) {
+	var refreshableCred = TemporaryCredential{}
+	var nextRefreshTime time.Time
+	for {
+		credentialProcessOutput, err := GenerateCredentials(&credentialsOptions)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// Assign credential values
+		refreshableCred.AccessKeyId = credentialProcessOutput.AccessKeyId
+		refreshableCred.SecretAccessKey = credentialProcessOutput.SecretAccessKey
+		refreshableCred.SessionToken = credentialProcessOutput.SessionToken // nosemgrep
+		refreshableCred.Expiration, _ = time.Parse(time.RFC3339, credentialProcessOutput.Expiration)
+		if (refreshableCred == TemporaryCredential{}) {
+			log.Println("no credentials created")
+			syscall.Exit(1)
+		}
+		// Get/create the credentials file path
+		readOnlyCredentialsFile, err := GetOrCreateReadOnlyCredentialsFile()
+		if err != nil {
+			log.Println("unable to get or create read-only AWS credentials file")
+			syscall.Exit(1)
+		}
+
+		// Read in all profiles in the credentials file
+		var lines []string
+		scanner := bufio.NewScanner(readOnlyCredentialsFile)
+		for scanner.Scan() {
+			lines = append(lines, scanner.Text())
+		}
+		readOnlyCredentialsFile.Close()
+
+		writeOnlyCredentialsFile, err := GetWriteOnlyCredentialsFile()
```

Comment:
If `writeOnlyCredentialsFile` is not being manipulated in this method, my thought is that it would stay inside of `WriteTo`.

Reply:
Good point - addressed in the next revision.